src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Triggers/UserTypeArgumentBindingProvider.cs

```diff
@@ -96,7 +97,14 @@ private static TInput GetBody(Message message, ValueBindingContext context)
                 }
                 else
                 {
-                    return message.GetBody<TInput>();
+                    try
+                    {
+                        return message.GetBody<TInput>();
+                    }
+                    catch (SerializationException e)
+                    {
+                        throw new InvalidOperationException($"XML formatted message is expected for ContentType='{message.ContentType}'. For JSON formatted message set ContentType='application/json': {e.Message} ");
```

Comment:
This is not XML formatted message, but DataContract XML serialized object. See https://github.com/Azure/azure-webjobs-sdk/issues/2154#issuecomment-494244856